lib/utils.rb

```diff
@@ -31,5 +32,31 @@ def Utils.paramhash_to_str params
   params.collect {|k, v| if (v == true) then ";#{k}" else ";#{k}=#{v}" end}.join("")
 end
 
+def Utils.check_route_matches_proxy route, proxy
+  regexp = /<sip:(.*?)@(.*?):(.*?);(.*?)>/i
```

Comment:
Also, you're not using $1 or $4, so you don't need to capture them with parentheses.